RUFAS/biophysical/field/soil/layer_data.py

```
@@ -429,18 +429,34 @@ def __post_init__(self, field_size: float, residue: float):
         if self.top_depth < 0 or self.bottom_depth <= 0 or self.top_depth >= self.bottom_depth:
+            OutputManager().add_error(
+                "Invalid layer values",
+                "Expected positive values for top and bottom depths of soil layer where top < bottom, "
+                f"received top: '{self.top_depth}', bottom: '{self.bottom_depth}'.",
+                info_map={"class": self.__class__.__name__, "function": self.__post_init__.__name__},
+            )
             raise ValueError(
-                f"Expected positive values for top and bottom depths of soil layer where top < bottom, "
+                "Expected positive values for top and bottom depths of soil layer where top < bottom, "
                 f"received top: '{self.top_depth}', bottom: '{self.bottom_depth}'."
             )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)